server/config/routes.rb

```diff
@@ -0,0 +1,25 @@
+Rails.application.routes.draw do
+  mount Rswag::Ui::Engine => '/api-docs'
+  mount Rswag::Api::Engine => '/api-docs'
+  get '/current_user', to: 'current_user#index'
+
+  devise_for :users, path: '/', path_names:
+  {
+    sign_in: 'login',
+    sign_out: 'logout',
+    registration: 'signup'
+  },
+                     controllers: {
+                       sessions: 'users/sessions',
+                       registrations: 'users/registrations'
+                     }
+
+  namespace :api do
+    namespace :v1 do
+      resources :city, only: %i[index create]
+      resources :homes, only: %i[index show create destroy]
+      resources :reservations, only: %i[index show create destroy]
+      resources :users, only: %i[:index, :show, :create]
```

Comment:
- [ ] When I try to register a new user from scratch, this happens:

![image](https://github.com/peterdtitan/BnB-homes/assets/57421823/0411f2c1-463c-44ce-8aea-c8115111530b)

This is because the route in the server does not exist for /api/v1/users:

![image](https://github.com/peterdtitan/BnB-homes/assets/57421823/bf70f516-9119-4680-ac11-03f43f7467c5)

![image](https://github.com/peterdtitan/BnB-homes/assets/57421823/a2d64db3-0468-4087-8f18-80e4e9ffe54a)


Also, as a result in localStorage, the user gets saved as "undefined".

If I try to login:

![image](https://github.com/peterdtitan/BnB-homes/assets/57421823/83b75302-965a-43b1-8b13-30a0a2376df9)

This is because "undefined" is not valid JSON.

## Let's solve the problem

### Can you notice the syntax error?

![image](https://github.com/peterdtitan/BnB-homes/assets/57421823/eca0b4af-8238-400d-93bf-2e13a570181f)

**Explanation:** _When you use i% before an array, it means "Non-interpolated Array of symbols, separated by whitespace".  You don't need the colons anymore.  Ruby now knows these elements are symbols.  If you add colons, you are effectively adding _double_ colons._

[Here](https://stackoverflow.com/questions/39803299/using-i-and-i-symbol-array-literal) is the complete explanation.

This seems the same syntax error that is present in `config/application.rb`.

Let's change it to `resources :users, only: %i[index show create]` without the colons.

After closing and running the server again, the routes are there!:

![image](https://github.com/peterdtitan/BnB-homes/assets/57421823/d273eeda-c3f9-48e7-94b1-5bb0fbe657f5)

I'll delete the "undefined" from localStorage and try again... 

_tip: You could do a guard clause with an if statement to prevent the "undefined" user to ever happen when the user is not successfully created:_

This is pseudocode to give you an idea:

```
if (user is created successfully without errors) {
    set user in localStorage
} else {
    don't do it! throw an error instead.
}
```

After correcting the routes and deleting the "undefined" user and trying to create a new one again:

![image](https://github.com/peterdtitan/BnB-homes/assets/57421823/40321ad8-21f9-447d-a886-be36d0d55e69)

It works!

Kindly, correct your routes bug as shown, and optionally, make sure the "undefined" user in localStorage does not ever exist again with an if guard clause, as shown. 💪